specifications/KeyValueStore/ClientCentric.tla

```diff
@@ -0,0 +1,185 @@
+--------------------------- MODULE ClientCentric ---------------------------
+EXTENDS Naturals, TLC, Sequences, FiniteSets, Util
+VARIABLES Keys, Values
+\* TODO InitValue could be bottom (_|_)
+
+\* TLA+ specifications of Client Centric Isolation Specification by Crooks et al: https://dl.acm.org/doi/10.1145/3087801.3087802
+\* TLA+ specifications by Tim Soethout (tim.soethout@ing.com)
+
+\* A database `State` is represented by keys with corresponding values
+State == [Keys -> Values]
+
+\* An `Operation` is a read or write of a key and value
+Operation == [op: {"read", "write"}, key: Keys, value: Values]
+
+\* Helpers representing Reads and Writes
+r(k,v) == [op |-> "read",  key |-> k, value |-> v]
+w(k,v) == [op |-> "write", key |-> k, value |-> v]
+
+\* A `Transaction` is a total order of `Operation`s
+\* Transaction == [ops: Seq(Operation), start: TimeStamp, commit: TimeStamp]
+Transaction == Seq(Operation)
+\* For simplicity we store start and commit in a lookup function
+TimeStamp == Nat
+TransactionTimes == [t \in Transaction |-> [start: TimeStamp, commit: TimeStamp]]
+
+\* "An execution e for a set of transactions
+\* T is a totally ordered set defined by the pair (Se,−−t \in T−→),
+\* where Se is the set of states generated by applying, 
+\* starting from the system’s initial state, a permutation of all the transactions in T ."
+ExecutionElem == [parentState: State, transaction: Transaction]
+\* resultState is the parentState of the next transaction, but not used in the isolation definitions.
+\* ExecutionElem == [parentState: State, transaction: Transaction, resultState: State]
+\* We represent an `Execution` as a sequence of `Transaction`s with their corresponding parent state.
+\* Note: This execution does therefor not contain the "final state" of the execution, since it is not a parent state of a transaction.
+Execution == Seq(ExecutionElem)
+
+\* Seq
+executionStates(execution) == [ i \in 1..Len(execution) |-> execution[i].parentState ]
+\* Set
+executionTransactions(execution) == { ep.transaction : ep \in SeqToSet(execution) }
+
+\* "The parent state is the last state in the execution
+\* Definition 1: s -T-> s' ≡ [(k,v) ∈ s ∧ (k,v)􏰍 \notin s'] => k ∈ W_T /\ w(k,v) ∈ Σ_T => (k,v) ∈ s.
```

Comment:
I can't tell if the box `[]` after `(k,v)` is a unicode rendering issue or part of the notation.

Reply:
Looks like it is a typo and unicode rendering issue.
https://github.com/cwi-swat/tla-ci/blob/53390b453ebbe6ed79a3f751863f5a4cdf32c5a0/ClientCentric.tla#L43C48-L43C50

It is in a comment, so it shouldn't matter. Feel free to remove it before merging.

Reply:
`􏰍` seems to be a copy/paste artifact from the original CI papers PDF. It can be removed.